music21/musicxml/xmlToM21.py

```diff
@@ -4536,59 +4547,25 @@ def xmlToChordSymbol(self, mxHarmony):
             if mxRootAlter is not None:
                 # can provide integer or float to create accidental on pitch
                 r.accidental = pitch.Accidental(float(mxRootAlter.text))
-            # set Pitch object on Harmony
-            cs.root(r)
+
+        # environLocal.printDebug(['mxToChordSymbol():', mxHarmony])
+        if mxFrame is not None:
+            cs = tablature.ChordWithFretBoard()
+        elif chordKind == 'none':
+            cs = harmony.NoChord(kindStr=chordKindStr)
         else:
+            cs = harmony.ChordSymbol(bass=b,
+                                     root=r,
+                                     inversion=inversion,
+                                     kind=chordKind,
+                                     kindStr=chordKindStr)
+
+        seta = _setAttributeFromTagText
+        if mxRoot is None:
             # function instead
             seta(cs, mxHarmony, 'function', 'romanNumeral')
 
-        if kindText is not None:  # two ways of doing it...
-            cs.chordKind = mxKind.text.strip()
-            # Get m21 chord kind from dict of musicxml aliases ("dominant" -> "dominant-seventh")
-            if cs.chordKind in harmony.CHORD_ALIASES:
-                cs.chordKind = harmony.CHORD_ALIASES[cs.chordKind]
-            mxKindText = mxKind.get('text')  # attribute
-            if mxKindText is not None:
-                if not ('NoChord' in cs.classes and mxKindText == ''):
-                    cs.chordKindStr = mxKindText
```

Comment:
moved above to 4528, but the NoChord condition had to be adjusted because we don't have an object yet